app/services/source_refresh_service.rb

```diff
@@ -0,0 +1,48 @@
+class SourceRefreshService
+  def initialize(source_id)
+    @source = Source.find(source_id)
+  end
+
+  def process
+    if @source.refresh_task_id.nil?
+      dispatch_refresh_upload_task
+    else
+      task = Task.find(@source.refresh_task_id)
+
+      if task.state == "completed"
+        if task.child_task_id.nil?
+          Rails.logger.error("PersisterTask is in creating, please try again later")
+          raise CatalogInventory::Api::Exception, "PersisterTask is in creating"
+        end
+
+        persister_task = Task.find(task.child_task_id)
+
+        if persister_task.state == "completed"
+          dispatch_refresh_upload_task
+        else
+          Rails.logger.error("PersisterTask #{persister_task.id} is running, please try again later")
+          raise CatalogInventory::Api::Exception, "PersisterTask #{persister_task.id} is running"
```

Comment:
@hsong-rh I think we should add a new exception which says we can run only refresh at a time.
CatalogInventory::Api::RefreshAlreadyRunningException